src/coreclr/scripts/superpmi_diffs_summarize.py

```diff
@@ -147,13 +147,86 @@ def main(main_args):
                     full_file_path = os.path.join(dirpath, file_name)
                     append_diff_file(f, arch, file_name, full_file_path, False)
 
-    print("##vso[task.uploadsummary]{}".format(final_md_path))
-
     with open(final_md_path, "r") as f:
         print(f.read())
 
+    # AzDO does not support syntax highlighting in fenced blocks. This hack rewrites ```diff blocks to HTML for AzDO purposes.
+    print("Coloring for AzDO...")
+    with open(final_md_path, "r") as f:
+        lines = f.read().splitlines()
+
+    inside_diff = False
+    cur_diff_lines = []
+    new_lines = []
+    for line in lines:
+        if line.startswith("```diff"):
+            inside_diff = True
+            cur_diff_lines = []
+        elif inside_diff and line.startswith("```"):
+            inside_diff = False
+            new_lines.append(html_color_diff(cur_diff_lines))
+        elif inside_diff:
+            cur_diff_lines.append(line)
+        else:
+            new_lines.append(line)
+
+    with open(final_md_path, "w") as f:
+        for line in new_lines:
+            f.write(line)
+            f.write("\n")
+
+    print("##vso[task.uploadsummary]{}".format(final_md_path))
+
     return 0
 
+def html_color_diff(lines):
+    new_text = ""
+
+    addition_line_color = "rgba(46,160,67,0.15)"
+    deletion_line_color = "rgba(248,81,73,0.15)"
+
+    cur_block = None
+    cur_block_color = None
+
+    def commit_block():
+        nonlocal new_text
+        nonlocal cur_block
+        nonlocal cur_block_color
+        if cur_block is None:
+            return
+
+        style = "display:block"
+
+        if cur_block_color is not None:
+            style = style + ";background-color:" + cur_block_color
+
+        new_block_text = '<span style="{}">'.format(style) + cur_block + "</span>"
```

Comment:
```suggestion
        new_block_text = '<div style="{}">'.format(style) + cur_block + "</div>"
```
before:
<img width="1389" alt="Screenshot 2023-02-19 at 22 49 57" src="https://user-images.githubusercontent.com/3840695/219974481-2e24df3e-a13f-41a3-9487-3f8955d5efd5.png">

after:
<img width="1411" alt="Screenshot 2023-02-19 at 22 51 14" src="https://user-images.githubusercontent.com/3840695/219974463-00ceddc5-94d7-4212-9b1e-b30a05139c85.png">

Reply:
The problem is that AzDO filters out block elements, it removes my `display:block` style from the spans. I have a fix where I just pad everything to the right length, but let me try to see if `div` works first or if it also filters that out.

Reply:
Nice, the div instead does the trick. Thank you!